delivery_postlogistics/tests/test_sanitize_values.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo.tests.common import SavepointCase
+
+from ..postlogistics.web_service import DISALLOWED_CHARS, PostlogisticsWebService
+
+ENDPOINT_URL = "https://wedecint.post.ch/"
+CLIENT_ID = "XXX"
+CLIENT_SECRET = "XXX"
+
+
+class TestSanitizeValues(SavepointCase):
+    """Just create records full of disallowed chars, an test that everyone of them
+    is correctly removed, as none of those chars should be sent on postlogistics api.
+    """
+
+    at_install = False
+    post_install = True
+
+    @classmethod
+    def setUpPartner(cls):
+        cls.partner = cls.env["res.partner"].create(
+            {
+                "name": "P<o\\t|at>o",
+                "mobile": "+33123456789>",
+                "phone": ">+33123456789<",
+                "email": "w<>|\\hatever@whatever.too",
+                "street": "42\\|<>whateverstraße",
+                "street2": "42\\|<>whateverstraße",
+                "zip": "43123\\",
+                "city": "Mouais\\<>|",
+            }
+        )
+
+    @classmethod
+    def setUpCarrier(cls):
+        shipping_product = cls.env["product.product"].create({"name": "Shipping"})
+        option_model = cls.env["postlogistics.delivery.carrier.template.option"]
+        partner_id = cls.env.ref("delivery_postlogistics.partner_postlogistics").id
+        label_layout = option_model.create({"code": "A6", "partner_id": partner_id})
+        output_format = option_model.create({"code": "PDF", "partner_id": partner_id})
+        image_resolution = option_model.create(
+            {"code": "600", "partner_id": partner_id}
+        )
+        cls.carrier = cls.env["delivery.carrier"].create(
+            {
+                "name": "Postlogistics",
+                "delivery_type": "postlogistics",
+                "product_id": shipping_product.id,
+                "postlogistics_endpoint_url": ENDPOINT_URL,
+                "postlogistics_client_id": CLIENT_ID,
+                "postlogistics_client_secret": CLIENT_SECRET,
+                "postlogistics_label_layout": label_layout.id,
+                "postlogistics_output_format": output_format.id,
+                "postlogistics_resolution": image_resolution.id,
+            }
+        )
```

Comment:
All this code is the same than the one in `test_postlogistics.py` (as well as `ENDPOINT_URL` etc), it deserves a common class both tests should inherit from.